src/lib/assets/yaml.ts

```diff
@@ -24,174 +26,154 @@ export async function overridesFile({ hash, name, image, config, apiToken }: Ass
+    env: genEnv(config, false, true),
+    image,
+    annotations: { "pepr.dev/description": config.description || "" },
+    labels: {
+      app: name,
+      "pepr.dev/controller": "admission",
+      "pepr.dev/uuid": config.uuid,
     },
+    securityContext: generateSecurityContext(),
+    readinessProbe: generateProbeConfig(),
+    livenessProbe: generateProbeConfig(),
+    resources: generateResourceConfig(),
+    containerSecurityContext: generateContainerSecurityContext(),
+    nodeSelector: {},
+    tolerations: [],
+    extraVolumeMounts: [],
+    extraVolumes: [],
+    affinity: {},
+    podAnnotations: {},
+    serviceMonitor: { enabled: false, labels: {}, annotations: {} },
   };
+}
 
-  await fs.writeFile(path, dumpYaml(overrides, { noRefs: true, forceQuotes: true }));
+// Generates the watcher object configuration
+export function generateWatcherConfig(config: Config, image: string, name: string) {
+  return {
+    terminationGracePeriodSeconds: 5,
+    env: genEnv(config, true, true),
+    image,
+    annotations: { "pepr.dev/description": `${config.description}` || "" },
+    labels: {
+      app: `${name}-watcher`,
+      "pepr.dev/controller": "watcher",
+      "pepr.dev/uuid": config.uuid,
+    },
+    securityContext: generateSecurityContext(),
+    readinessProbe: generateProbeConfig(),
+    livenessProbe: generateProbeConfig(),
+    resources: generateResourceConfig(),
+    containerSecurityContext: generateContainerSecurityContext(),
+    nodeSelector: {},
+    tolerations: [],
+    extraVolumeMounts: [],
+    extraVolumes: [],
+    affinity: {},
+    podAnnotations: {},
+    serviceMonitor: { enabled: false, labels: {}, annotations: {} },
+  };
 }
-export function zarfYaml({ name, image, config }: Assets, path: string) {
-  const zarfCfg = {
-    kind: "ZarfPackageConfig",
-    metadata: {
-      name,
-      description: `Pepr Module: ${config.description}`,
-      url: "https://github.com/defenseunicorns/pepr",
-      version: `${config.appVersion || "0.0.1"}`,
+
+// Generates the security context configuration
+export function generateSecurityContext() {
+  return {
+    runAsUser: 65532,
+    runAsGroup: 65532,
+    runAsNonRoot: true,
+    fsGroup: 65532,
+  };
+}
+
+// Generates the container security context
+export function generateContainerSecurityContext() {
+  return {
+    runAsUser: 65532,
+    runAsGroup: 65532,
+    runAsNonRoot: true,
+    allowPrivilegeEscalation: false,
+    capabilities: {
+      drop: ["ALL"],
     },
-    components: [
-      {
-        name: "module",
-        required: true,
-        manifests: [
-          {
-            name: "module",
-            namespace: "pepr-system",
-            files: [path],
-          },
-        ],
-        images: [image],
-      },
-    ],
   };
+}
 
-  return dumpYaml(zarfCfg, { noRefs: true });
+// Generates the probe configuration
+export function generateProbeConfig() {
```

Comment:
Clear function name, could use a return type
```suggestion
export function generateProbeConfig() {
```

Reply:
Done